### PR TITLE
Add dimension and consistency checks for  size / corners and num_cells / dx

### PR DESCRIPTION
**Pull request by streeve**
Also adds errors if neither are available

### CODE REVIEW

File: src/CabanaPD_Input.hpp [hunk 1]
```diff
@@ -104,7 +89,73 @@ class Inputs
...
+            }
+            inputs["low_corner"]["unit"] = size_unit;
+            inputs["high_corner"]["unit"] = size_unit;
+        }
+        else if ( inputs.contains( "low_corner" ) &&
+                  ( inputs.contains( "high_corner" ) ) )
+        {
+            auto low_corner = inputs["low_corner"]["value"];
+            auto high_corner = inputs["high_corner"]["value"];
+            std::string size_unit = inputs["low_corner"]["unit"];
```
Comment (pabloseleson) [inline src/CabanaPD_Input.hpp:115]: It is possible that the user accidentally includes a different unit for low_corner and high_corner. Should we output an error in that case?

File: src/CabanaPD_Input.hpp [hunk 2]
```diff
@@ -104,7 +89,73 @@ class Inputs
...
+            for ( std::size_t d = 0; d < low_corner.size(); d++ )
```
Comment (pabloseleson) [inline src/CabanaPD_Input.hpp:116]: It is possible that the user accidentally inputs arrays of different dimensions for low_corner and high_corner. Should we output an error in that case?

File: src/CabanaPD_Input.hpp [hunk 3]
```diff
@@ -104,7 +89,73 @@ class Inputs
...
+            {
+                double low = low_corner[d];
+                double high = high_corner[d];
+                inputs["system_size"]["value"][d] = high - low;
+            }
+            inputs["system_size"]["unit"] = size_unit;
+        }
+        else
+        {
+            throw std::runtime_error( "Must input either system_size or "
```
Comment (pabloseleson) [inline src/CabanaPD_Input.hpp:126]: Are the quotes " " in: 'or " "both' OK?

File: src/CabanaPD_Input.hpp [hunk 4]
```diff
@@ -104,7 +89,73 @@ class Inputs
...
+                                      "both low_corner and high_corner." );
+        }
+
+        if ( inputs.contains( "dx" ) )
+        {
+            auto size = inputs["system_size"]["value"];
+            for ( std::size_t d = 0; d < size.size(); d++ )
+            {
+                double system_size = inputs["system_size"]["value"][d];
```
Comment (pabloseleson) [inline src/CabanaPD_Input.hpp:135]: Could we use instead the following?
double system_size = size[d];?

File: src/CabanaPD_Input.hpp [hunk 5]
```diff
@@ -104,7 +89,73 @@ class Inputs
...
+                inputs["num_cells"]["value"][d] =
+                    static_cast<int>( system_size / dx );
+            }
+        }
+        else if ( inputs.contains( "num_cells" ) )
+        {
+            auto size = inputs["system_size"]["value"];
+            for ( std::size_t d = 0; d < size.size(); d++ )
+            {
+                double low = inputs["low_corner"]["value"][d];
```
Comment (pabloseleson) [inline src/CabanaPD_Input.hpp:146]: Can we replace these lines 

  double low = inputs["low_corner"]["value"][d]; 
  double high = inputs["low_corner"]["value"][d];
  double nc = inputs["num_cells"]["value"][d];
  inputs["dx"]["value"][d] = ( high - low ) / nc;

by the following?

  double system_size = size[d];
  double nc = inputs["num_cells"]["value"][d];
  inputs["dx"]["value"][d] = system_size / nc;

File: src/CabanaPD_Input.hpp (hunk 1 above)
Comment (streeve) [reply src/CabanaPD_Input.hpp:124]: Added

File: src/CabanaPD_Input.hpp (hunk 2 above)
Comment (streeve) [reply src/CabanaPD_Input.hpp]: Added that it is currently just 3d (we are close to a possibility of 2d)

File: src/CabanaPD_Input.hpp (hunk 3 above)
Comment (streeve) [reply src/CabanaPD_Input.hpp:135]: Yes. The CI would catch that if it were not legal syntax

File: src/CabanaPD_Input.hpp (hunk 4 above)
Comment (streeve) [reply src/CabanaPD_Input.hpp]: Fixed

File: src/CabanaPD_Input.hpp [hunk 6]
```diff
@@ -104,7 +89,73 @@ class Inputs
...
+                                      "both low_corner and high_corner." );
+        }
+
+        if ( inputs.contains( "dx" ) )
+        {
+            auto size = inputs["system_size"]["value"];
+            for ( std::size_t d = 0; d < size.size(); d++ )
+            {
+                double system_size = inputs["system_size"]["value"][d];
+                double dx = inputs["dx"]["value"][d];
```
[the remark this answers is not in the document]
Comment (streeve) [reply src/CabanaPD_Input.hpp]: Added

File: src/CabanaPD_Input.hpp (hunk 5 above)
Comment (streeve) [reply src/CabanaPD_Input.hpp]: Good point, fixed

**Review by streeve — COMMENTED** [review]
@pabloseleson can you test a couple examples to make sure this is working?

**Comment by streeve** [comment]
@pabloseleson this now only addresses your requested changes since it was needed in #155

File: src/CabanaPD_Input.hpp [hunk 7]
```diff
@@ -110,10 +113,16 @@ class Inputs
         else if ( inputs.contains( "low_corner" ) &&
                   ( inputs.contains( "high_corner" ) ) )
```
Comment (pabloseleson) [inline src/CabanaPD_Input.hpp:114]: It seems we have here extra parentheses: 
"( inputs.contains( "high_corner" ) ) )" could be "inputs.contains( "high_corner" ) )"

File: src/CabanaPD_Input.hpp [hunk 8]
```diff
@@ -95,6 +95,9 @@ class Inputs
         if ( inputs.contains( "system_size" ) )
         {
             auto size = inputs["system_size"]["value"];
+            if ( size != 3 )
```
Comment (pabloseleson) [inline src/CabanaPD_Input.hpp:98]: As opposed to Line 116, size here is not the size of an array but the size of the system. So, I believe "if ( size != 3 )" is wrong.

File: src/CabanaPD_Input.hpp [hunk 9]
```diff
@@ -110,10 +113,16 @@ class Inputs
...
         {
+            auto size = inputs["low_corner"]["value"].size();
+            if ( size != 3 )
+                log_err( std::cout, "CabanaPD requires 3d (low_corner)." );
+            if ( inputs["high_corner"]["value"].size() != 3 )
+                log_err( std::cout, "CabanaPD requires 3d (high_corner)." );
+
             auto low_corner = inputs["low_corner"]["value"];
             auto high_corner = inputs["high_corner"]["value"];
             std::string size_unit = inputs["low_corner"]["unit"];
```
Comment (pabloseleson) [inline src/CabanaPD_Input.hpp:124]: "string size_unit" is confusing here, because "size" is the array dimension not the domain size.

**Comment by pabloseleson** [comment]
Here are 3 tests with outputs:

**Test 1:** 

Inputs:

    "num_cells"              : {"value": [10, 20, 25]},
    "system_size"            : {"value": [100.0, 150.0, 300.0], "unit": "m"},

Outputs:

 =========== 
 system_size : (100.0 , 150.0 , 300.0) "m"
 low_corner  : (-50.0 , -75.0 , -150.0) "m"
 high_corner : (50.0 , 75.0 , 150.0) "m"
 =========== 
 nc          : (10 , 20 , 25) 
 dx          : (10.0 , 7.5 , 12.0) "m"

**Test 2:**

Inputs:

    "dx"                     : {"value": [10.0, 7.5, 12.0], "unit": "m"},
    "low_corner"            : {"value":  [-50.0 , -75.0 , -150.0], "unit": "m"},
    "high_corner"            : {"value": [50.0 , 75.0 , 150.0], "unit": "m"},

Outputs:

 =========== 
 system_size : (100.0 , 150.0 , 300.0) "m"
 low_corner  : (-50.0 , -75.0 , -150.0) "m"
 high_corner : (50.0 , 75.0 , 150.0) "m"
 =========== 
 nc          : (10 , 20 , 25) 
 dx          : (10.0 , 7.5 , 12.0) "m"

**Test 3:**

Inputs:

    "dx"                     : {"value": [10.0, 7.5, 12.0], "unit": "m"},
    "low_corner"            : {"value":  [-25.0 , 0.0 , -75.0], "unit": "m"},
    "high_corner"            : {"value": [75.0 , 150.0 , 225.0], "unit": "m"},

Outputs: 

 =========== 
 system_size : (100.0 , 150.0 , 300.0) "m"
 low_corner  : (-25.0 , 0.0 , -75.0) "m"
 high_corner : (75.0 , 150.0 , 225.0) "m"
 =========== 
 nc          : (10 , 20 , 25) 
 dx          : (10.0 , 7.5 , 12.0) "m"

Output code

        std::cout << " =========== " << std::endl;
        std::cout << " system_size : " << "(" << inputs["system_size"]["value"][0] << " , " << inputs["system_size"]["value"][1] << " , " << inputs["system_size"]["value"][2] << ") " << inputs["system_size"]["unit"] << std::endl;
        std::cout << " low_corner  : " << "(" << inputs["low_corner"]["value"][0] << " , " << inputs["low_corner"]["value"][1] << " , " << inputs["low_corner"]["value"][2] << ") " << inputs["low_corner"]["unit"] << std::endl;
        std::cout << " high_corner : " << "(" << inputs["high_corner"]["value"][0] << " , " << inputs["high_corner"]["value"][1] << " , " << inputs["high_corner"]["value"][2] << ") " << inputs["high_corner"]["unit"] << std::endl;
        std::cout << " =========== " << std::endl;
        std::cout << " nc          : " << "(" << inputs["num_cells"]["value"][0] << " , " << inputs["num_cells"]["value"][1] << " , " << inputs["num_cells"]["value"][2] << ") " << std::endl;
        std::cout << " dx          : " << "(" << inputs["dx"]["value"][0] << " , " << inputs["dx"]["value"][1] << " , " << inputs["dx"]["value"][2] << ") " << inputs["dx"]["unit"] << std::endl;

**Comment by pabloseleson** [comment]
Tests for inconsistencies:

**Test i**

Inputs:

    "dx"                     : {"value": [10.0, 7.5, 12.0], "unit": "cm"},
    "low_corner"            : {"value":  [-25.0 , 0.0 , -75.0], "unit": "m"},
    "high_corner"            : {"value": [75.0 , 150.0 , 225.0], "unit": "m"},

Output

Units for dx do not match system units.
libc++abi: terminating due to uncaught exception of type std::runtime_error: Aborting after error from input. See error file.
[pscstudio:65566] *** Process received signal ***
[pscstudio:65566] Signal: Abort trap: 6 (6)
[pscstudio:65566] Signal code:  (0)
[pscstudio:65566] [ 0] 0   libsystem_platform.dylib            0x0000000193146584 _sigtramp + 56
[pscstudio:65566] [ 1] 0   libsystem_pthread.dylib             0x0000000193115c20 pthread_kill + 288
[pscstudio:65566] [ 2] 0   libsystem_c.dylib                   0x0000000193022a30 abort + 180
[pscstudio:65566] [ 3] 0   libc++abi.dylib                     0x00000001930ccd08 _ZN10__cxxabiv130__aligned_malloc_with_fallbackEm + 0
[pscstudio:65566] [ 4] 0   libc++abi.dylib                     0x00000001930bcfa4 _ZL28demangling_terminate_handlerv + 320
[pscstudio:65566] [ 5] 0   libobjc.A.dylib                     0x0000000192d57c00 _ZL15_objc_terminatev + 160
[pscstudio:65566] [ 6] 0   libc++abi.dylib                     0x00000001930cc0cc _ZSt11__terminatePFvvE + 16
[pscstudio:65566] [ 7] 0   libc++abi.dylib                     0x00000001930cf348 __cxa_get_exception_ptr + 0
[pscstudio:65566] [ 8] 0   libc++abi.dylib                     0x00000001930cf28c _ZN10__cxxabiv1L22exception_cleanup_funcE19_Unwind_Reason_CodeP17_Unwind_Exception + 0
[pscstudio:65566] [ 9] 0   ElasticWave                         0x000000010290f718 _ZN8CabanaPD7log_errINSt3__113basic_ostreamIcNS1_11char_traitsIcEEEERA40_KcEEvRT_OT0_ + 140
[pscstudio:65566] [10] 0   ElasticWave                         0x00000001029022a0 _ZN8CabanaPD6Inputs9setupSizeEv + 3800
[pscstudio:65566] [11] 0   ElasticWave                         0x00000001029007cc _ZN8CabanaPD6InputsC2ENSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEE + 128
[pscstudio:65566] [12] 0   ElasticWave                         0x00000001028fe624 _Z18elasticWaveExampleNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEE + 140
[pscstudio:65566] [13] 0   ElasticWave                         0x00000001029006f8 main + 184
[pscstudio:65566] [14] 0   dyld                                0x0000000192d8b154 start + 2476
[pscstudio:65566] *** End of error message ***
zsh: abort      ./ElasticWave

File: src/CabanaPD_Input.hpp (hunk 9 above)
Comment (streeve) [reply src/CabanaPD_Input.hpp]: Changed

**Comment by pabloseleson** [comment]
Tests for inconsistent dimensions

**Test a** 

Inputs:

    "dx"                     : {"value": [10.0, 7.5], "unit": "m"},
    "system_size"            : {"value": [100.0, 150.0, 300.0], "unit": "m"},

Output:

CabanaPD requires 3d (dx).
libc++abi: terminating due to uncaught exception of type std::runtime_error: Aborting after error from input. See error file.
[pscstudio:67717] *** Process received signal ***
[pscstudio:67717] Signal: Abort trap: 6 (6)
[pscstudio:67717] Signal code:  (0)
[pscstudio:67717] [ 0] 0   libsystem_platform.dylib            0x0000000193146584 _sigtramp + 56
[pscstudio:67717] [ 1] 0   libsystem_pthread.dylib             0x0000000193115c20 pthread_kill + 288
[pscstudio:67717] [ 2] 0   libsystem_c.dylib                   0x0000000193022a30 abort + 180
[pscstudio:67717] [ 3] 0   libc++abi.dylib                     0x00000001930ccd08 _ZN10__cxxabiv130__aligned_malloc_with_fallbackEm + 0
[pscstudio:67717] [ 4] 0   libc++abi.dylib                     0x00000001930bcfa4 _ZL28demangling_terminate_handlerv + 320
[pscstudio:67717] [ 5] 0   libobjc.A.dylib                     0x0000000192d57c00 _ZL15_objc_terminatev + 160
[pscstudio:67717] [ 6] 0   libc++abi.dylib                     0x00000001930cc0cc _ZSt11__terminatePFvvE + 16
[pscstudio:67717] [ 7] 0   libc++abi.dylib                     0x00000001930cf348 __cxa_get_exception_ptr + 0
[pscstudio:67717] [ 8] 0   libc++abi.dylib                     0x00000001930cf28c _ZN10__cxxabiv1L22exception_cleanup_funcE19_Unwind_Reason_CodeP17_Unwind_Exception + 0
[pscstudio:67717] [ 9] 0   ElasticWave                         0x000000010415b538 _ZN8CabanaPD7log_errINSt3__113basic_ostreamIcNS1_11char_traitsIcEEEERA27_KcEEvRT_OT0_ + 140
[pscstudio:67717] [10] 0   ElasticWave                         0x000000010414dd94 _ZN8CabanaPD6Inputs9setupSizeEv + 2508
[pscstudio:67717] [11] 0   ElasticWave                         0x000000010414c7cc _ZN8CabanaPD6InputsC2ENSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEE + 128
[pscstudio:67717] [12] 0   ElasticWave                         0x000000010414a624 _Z18elasticWaveExampleNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEE + 140
[pscstudio:67717] [13] 0   ElasticWave                         0x000000010414c6f8 main + 184
[pscstudio:67717] [14] 0   dyld                                0x0000000192d8b154 start + 2476
[pscstudio:67717] *** End of error message ***
zsh: abort      ./ElasticWave 

**Test b**

Inputs:

    "dx"                     : {"value": [10.0, 7.5, 12.0], "unit": "m"},
    "system_size"            : {"value": [100.0, 150.0], "unit": "m"},

Output:

CabanaPD requires 3d (system_size).
libc++abi: terminating due to uncaught exception of type std::runtime_error: Aborting after error from input. See error file.
[pscstudio:69243] *** Process received signal ***
[pscstudio:69243] Signal: Abort trap: 6 (6)
[pscstudio:69243] Signal code:  (0)
[pscstudio:69243] [ 0] 0   libsystem_platform.dylib            0x0000000193146584 _sigtramp + 56
[pscstudio:69243] [ 1] 0   libsystem_pthread.dylib             0x0000000193115c20 pthread_kill + 288
[pscstudio:69243] [ 2] 0   libsystem_c.dylib                   0x0000000193022a30 abort + 180
[pscstudio:69243] [ 3] 0   libc++abi.dylib                     0x00000001930ccd08 _ZN10__cxxabiv130__aligned_malloc_with_fallbackEm + 0
[pscstudio:69243] [ 4] 0   libc++abi.dylib                     0x00000001930bcfa4 _ZL28demangling_terminate_handlerv + 320
[pscstudio:69243] [ 5] 0   libobjc.A.dylib                     0x0000000192d57c00 _ZL15_objc_terminatev + 160
[pscstudio:69243] [ 6] 0   libc++abi.dylib                     0x00000001930cc0cc _ZSt11__terminatePFvvE + 16
[pscstudio:69243] [ 7] 0   libc++abi.dylib                     0x00000001930cf348 __cxa_get_exception_ptr + 0
[pscstudio:69243] [ 8] 0   libc++abi.dylib                     0x00000001930cf28c _ZN10__cxxabiv1L22exception_cleanup_funcE19_Unwind_Reason_CodeP17_Unwind_Exception + 0
[pscstudio:69243] [ 9] 0   ElasticWave                         0x00000001024073f8 _ZN8CabanaPD7log_errINSt3__113basic_ostreamIcNS1_11char_traitsIcEEEERA36_KcEEvRT_OT0_ + 140
[pscstudio:69243] [10] 0   ElasticWave                         0x00000001023f9628 _ZN8CabanaPD6Inputs9setupSizeEv + 608
[pscstudio:69243] [11] 0   ElasticWave                         0x00000001023f87cc _ZN8CabanaPD6InputsC2ENSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEE + 128
[pscstudio:69243] [12] 0   ElasticWave                         0x00000001023f6624 _Z18elasticWaveExampleNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEE + 140
[pscstudio:69243] [13] 0   ElasticWave                         0x00000001023f86f8 main + 184
[pscstudio:69243] [14] 0   dyld                                0x0000000192d8b154 start + 2476
[pscstudio:69243] *** End of error message ***
zsh: abort      ./ElasticWave 

**Test c** 

Inputs: 

    "dx"                     : {"value": [10.0, 7.5, 12.0], "unit": "m"},
    "low_corner"            : {"value":  [-25.0 , 0.0 ], "unit": "m"},
    "high_corner"            : {"value": [75.0 , 150.0 , 225.0], "unit": "m"},

Output: 

CabanaPD requires 3d (low_corner).
libc++abi: terminating due to uncaught exception of type std::runtime_error: Aborting after error from input. See error file.
[pscstudio:69508] *** Process received signal ***
[pscstudio:69508] Signal: Abort trap: 6 (6)
[pscstudio:69508] Signal code:  (0)
[pscstudio:69508] [ 0] 0   libsystem_platform.dylib            0x0000000193146584 _sigtramp + 56
[pscstudio:69508] [ 1] 0   libsystem_pthread.dylib             0x0000000193115c20 pthread_kill + 288
[pscstudio:69508] [ 2] 0   libsystem_c.dylib                   0x0000000193022a30 abort + 180
[pscstudio:69508] [ 3] 0   libc++abi.dylib                     0x00000001930ccd08 _ZN10__cxxabiv130__aligned_malloc_with_fallbackEm + 0
[pscstudio:69508] [ 4] 0   libc++abi.dylib                     0x00000001930bcfa4 _ZL28demangling_terminate_handlerv + 320
[pscstudio:69508] [ 5] 0   libobjc.A.dylib                     0x0000000192d57c00 _ZL15_objc_terminatev + 160
[pscstudio:69508] [ 6] 0   libc++abi.dylib                     0x00000001930cc0cc _ZSt11__terminatePFvvE + 16
[pscstudio:69508] [ 7] 0   libc++abi.dylib                     0x00000001930cf348 __cxa_get_exception_ptr + 0
[pscstudio:69508] [ 8] 0   libc++abi.dylib                     0x00000001930cf28c _ZN10__cxxabiv1L22exception_cleanup_funcE19_Unwind_Reason_CodeP17_Unwind_Exception + 0
[pscstudio:69508] [ 9] 0   ElasticWave                         0x00000001005eb498 _ZN8CabanaPD7log_errINSt3__113basic_ostreamIcNS1_11char_traitsIcEEEERA35_KcEEvRT_OT0_ + 140
[pscstudio:69508] [10] 0   ElasticWave                         0x00000001005dd914 _ZN8CabanaPD6Inputs9setupSizeEv + 1356
[pscstudio:69508] [11] 0   ElasticWave                         0x00000001005dc7cc _ZN8CabanaPD6InputsC2ENSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEE + 128
[pscstudio:69508] [12] 0   ElasticWave                         0x00000001005da624 _Z18elasticWaveExampleNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEE + 140
[pscstudio:69508] [13] 0   ElasticWave                         0x00000001005dc6f8 main + 184
[pscstudio:69508] [14] 0   dyld                                0x0000000192d8b154 start + 2476
[pscstudio:69508] *** End of error message ***
zsh: abort      ./ElasticWave 

**Test d**

Inputs:

    "dx"                     : {"value": [10.0, 7.5, 12.0], "unit": "m"},
    "low_corner"            : {"value":  [-25.0 , 0.0 , -75.0], "unit": "m"},
    "high_corner"            : {"value": [75.0 , 150.0 ], "unit": "m"},

Output:

CabanaPD requires 3d (high_corner).
libc++abi: terminating due to uncaught exception of type std::runtime_error: Aborting after error from input. See error file.
[pscstudio:69783] *** Process received signal ***
[pscstudio:69783] Signal: Abort trap: 6 (6)
[pscstudio:69783] Signal code:  (0)
[pscstudio:69783] [ 0] 0   libsystem_platform.dylib            0x0000000193146584 _sigtramp + 56
[pscstudio:69783] [ 1] 0   libsystem_pthread.dylib             0x0000000193115c20 pthread_kill + 288
[pscstudio:69783] [ 2] 0   libsystem_c.dylib                   0x0000000193022a30 abort + 180
[pscstudio:69783] [ 3] 0   libc++abi.dylib                     0x00000001930ccd08 _ZN10__cxxabiv130__aligned_malloc_with_fallbackEm + 0
[pscstudio:69783] [ 4] 0   libc++abi.dylib                     0x00000001930bcfa4 _ZL28demangling_terminate_handlerv + 320
[pscstudio:69783] [ 5] 0   libobjc.A.dylib                     0x0000000192d57c00 _ZL15_objc_terminatev + 160
[pscstudio:69783] [ 6] 0   libc++abi.dylib                     0x00000001930cc0cc _ZSt11__terminatePFvvE + 16
[pscstudio:69783] [ 7] 0   libc++abi.dylib                     0x00000001930cf348 __cxa_get_exception_ptr + 0
[pscstudio:69783] [ 8] 0   libc++abi.dylib                     0x00000001930cf28c _ZN10__cxxabiv1L22exception_cleanup_funcE19_Unwind_Reason_CodeP17_Unwind_Exception + 0
[pscstudio:69783] [ 9] 0   ElasticWave                         0x00000001048ab3f8 _ZN8CabanaPD7log_errINSt3__113basic_ostreamIcNS1_11char_traitsIcEEEERA36_KcEEvRT_OT0_ + 140
[pscstudio:69783] [10] 0   ElasticWave                         0x000000010489d984 _ZN8CabanaPD6Inputs9setupSizeEv + 1468
[pscstudio:69783] [11] 0   ElasticWave                         0x000000010489c7cc _ZN8CabanaPD6InputsC2ENSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEE + 128
[pscstudio:69783] [12] 0   ElasticWave                         0x000000010489a624 _Z18elasticWaveExampleNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEE + 140
[pscstudio:69783] [13] 0   ElasticWave                         0x000000010489c6f8 main + 184
[pscstudio:69783] [14] 0   dyld                                0x0000000192d8b154 start + 2476
[pscstudio:69783] *** End of error message ***
zsh: abort      ./ElasticWave 

**Test e**

Inputs: 

    "num_cells"              : {"value": [10, 20]},
    "system_size"            : {"value": [100.0, 150.0, 300.0], "unit": "m"},

Output: 

CabanaPD requires 3d (num_cells).
...

**Test f**

Inputs: 

    "num_cells"              : {"value": [10, 20, 25]},
    "system_size"            : {"value": [100.0, 150.0], "unit": "m"},

Output: 

CabanaPD requires 3d (system_size).
...

**Review by pabloseleson — APPROVED** [review]

**Pull request merged**